ethereum/core/src/main/java/tech/pegasys/teku/core/operationsignatureverifiers/ProposerSlashingSignatureVerifier.java

```diff
@@ -40,11 +39,7 @@ public boolean verifySignature(
     final BeaconBlockHeader header1 = proposerSlashing.getHeader_1().getMessage();
     final BeaconBlockHeader header2 = proposerSlashing.getHeader_2().getMessage();
     BLSPublicKey publicKey =
-        BeaconStateCache.getTransitionCaches(state)
-            .getValidatorsPubKeys()
-            .get(
-                header1.getProposerIndex(),
-                idx -> state.getValidators().get(toIntExact(idx.longValue())).getPubkey());
+        ValidatorsUtil.getValidatorPubKey(state, header1.getProposerIndex()).orElseThrow();
```

Comment:
nit: Would be good to just return `false` here if the pub key isn't available. Little surprised we've gotten away with the current version actually as we don't appear to check the proposer index ahead of time.

Reply:
Done. Looks like the slashing index is checked just here (i.e. when validating gossiped slashing) : https://github.com/ConsenSys/teku/blob/ae994ca4538eba3d7063fb4990277385f75b26c2/ethereum/statetransition/src/main/java/tech/pegasys/teku/statetransition/validation/ProposerSlashingValidator.java#L77
But is not checked when validating block slashings

Reply:
Created the issue on this: https://github.com/ConsenSys/teku/issues/3319